File: examples/python/subscribe_candlesticks.py
from time import sleep

from longbridge.openapi import (
    QuoteContext,
    Config,
    OAuthBuilder,
    Period,
    PushCandlestick,
    TradeSessions,
)


def on_candlestick(symbol: str, event: PushCandlestick):
    print(symbol, event)


oauth = OAuthBuilder("your-client-id").build(
    lambda url: print(f"Open this URL to authorize: {url}")
)
config = Config.from_oauth(oauth)
ctx = QuoteContext(config)
ctx.set_on_candlestick(on_candlestick)
ctx.subscribe_candlesticks(
    "AAPL.US",
    Period.Min_1,
    TradeSessions.Intraday,
)
sleep(30)

File: examples/nodejs/subscribe_candlesticks.js
const { Config, QuoteContext, Period, TradeSessions, OAuth } = require('longbridge');

let globalCtx;

async function main() {
  const oauth = await OAuth.build("your-client-id", (_, url) => {
    console.log("Open this URL to authorize: " + url);
  });
  let config = Config.fromOAuth(oauth);
  globalCtx = QuoteContext.new(config);
  globalCtx.setOnCandlestick((_, event) => console.log(event.toString()));
  await globalCtx.subscribeCandlesticks(
    "AAPL.US",
    Period.Min_1,
    TradeSessions.Intraday
  );
}

Promise.all([main()]).catch((err) => console.error(err));
